plugins/modules/win_dhcp_lease.ps1

```diff
@@ -0,0 +1,442 @@
+                $module.Result.lease = Convert-ReturnValue -Object $reservation
+            } else {
+                $module.Result.changed = $true
+            }
+
+            # Return values
+            $module.ExitJson()
+        }
+    }
+
+    # Lease Doesn't Exist - Create
+    if ($current_lease_exists -eq $false) {
+        # Required: Scope ID
+        if (-not $scope_id) {
+            $module.Result.changed = $false
+            $module.FailJson("The scope_id parameter is required for state=present when a lease or reservation doesn't already exist")
+        }
+
+        # Required Parameters
+        $lease_params = @{
+            ClientId = $mac
+            IPAddress = $ip
+            ScopeId = $scope_id
+            AddressState = 'Active'
+            Confirm = $false
+        }
+
+        if ($duration) {
+            $lease_params.LeaseExpiryTime = (Get-Date).AddDays($duration)
+        }
+
+        if ($dns_hostname) {
+            $lease_params.HostName = $dns_hostname
+        }
+
+        if ($dns_regtype) {
+            $lease_params.DnsRR = $dns_regtype
+        }
+
+        if ($description) {
+            $lease_params.Description = $description
+        }
+
+        # Create Lease
+        Try {
+            # Create lease based on parameters
+            Add-DhcpServerv4Lease @lease_params -WhatIf:$check_mode
+
+            # Retreive the lease
+            if(-not $check_mode) {
+                $new_lease = Get-DhcpServerv4Lease -ClientId $mac -ScopeId $scope_id
+                $module.Result.lease = Convert-ReturnValue -Object $new_lease
+            }
+
+            # If lease is the desired type
+            if ($type -eq "lease") {
+                $module.Result.changed = $true
+                $module.ExitJson()
+            }
+        }
+        Catch {
+            # Failed to create lease
+            $module.FailJson("Could not create DHCP lease",$_)
+        }
+
+        # Create Reservation
+        Try {
+            # If reservation is the desired type
+            if ($type -eq "reservation") {
+                if ($reservation_name) {
+                    $lease_params.Name = $reservation_name
+                }
+                else {
+                    $lease_params.Name = "reservation-" + $mac
+                }
+
+                Try {
+                    if($check_mode) {
+                        # In check mode, a lease won't exist for conversion, make one manually
+                        Add-DhcpServerv4Reservation -ScopeId $scope_id -ClientId $mac_original -IPAddress $ip -WhatIf:$check_mode
+                    } else {
+                        # Convert to Reservation
+                        $new_lease | Add-DhcpServerv4Reservation -WhatIf:$check_mode
+                    }
+                }
+                Catch {
+                    # Failed to create reservation
+                    $module.FailJson("Could not create DHCP reservation",$_)
```

Comment:
It would be nice to at least keep the exception message for non `-vvv` users

```suggestion
                    $module.FailJson("Could not create DHCP reservation: $($_.Exception.Message)", $_)
```